frontend/apps/core-app/src/hooks/useAPICall.ts

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+type Parameter<T> = T extends (...args: infer T) => any ? T : never;
+type AsyncReturnType<T extends (...args: any) => any> = T extends (
```

Comment:
didn't we at some point say that api/client functions would take their parameters wrapped in an object? might just have been in my head. but that way, it would work.

Reply:
I'm not sure if we made a decision on that, but that would tie us to a specific way of writing function. This lets the api call work with any function.

For example our get recipient client method was:
```
get = async ({
    recordId,
    formId,
    databaseId,
  }: RecordLookup): Promise<FormWithRecord<Recipient>[]>
```

I think our list recipient endpoint could look like:
```
list = async ({
    formId,
    databaseId,
  }: FormLookup, options?: PagingOptions): Promise<FormWithRecord<Recipient>[]>
```

Reply:
I don't really see how your example supports your argument, both cases have their arguments wrapped in objects, but since your argument is about functionality, and mine is mostly about readability and code style, I'd say yours is stronger.

Reply:
In the list example there are two arguments to the function, `FormLookup` and `PagingOptions`. If we turned `args` from an array to a single value then we wouldn't be able to support both arguments.